### PR TITLE
overhaul developer docs

### PR DESCRIPTION
**Pull request by dwhieb**
This is a major overhaul of the developer documentation (CONTRIBUTING.md) that should make it significantly easier to get started on this project.

@vadekh @TongJ05, could you each look at this, let me know if it makes sense, and let me know if you think there's any information that's missing or could be improved? Thanks!

(@vadekh I can't add you as a second reviewer while this repo is still private, but consider yourself added haha.)

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -48,153 +66,481 @@ This project is driven by the needs of documentary and descriptive linguists, an
...
+});
+```
+
+**NOTE:** Some unit tests have to be run in the browser because they rely on specific browser APIs (for example, the tests for `Model.js`). These should be given the `.component.js` extension so that they run in the browser, but otherwise can be written as normal unit tests.
+
+#### Component Tests
+
+Component tests check the functionality of a single component in isolation. Any items in the `components/` folder that have functionality, or any page-specific components in `pages/{ComponentName}` that have functionality, should have a component test.
+
+Component tests are run using [Cypress] tests on a single [Storybook] story. To load the component, call `cy.visit()` on the URL for that particular story. You can find the URL by opening that story in Storybook,clicking the "Open canvas in new tab" icon in the upper right (the external link icon), and copying the URL from that tab. Now you can write your component tests using Cypress' API. Note that you do *not* need to import Chai or Sinon—these are included in Cypress. Here is a small example test:
```
Comment (vadekh) [inline .github/CONTRIBUTING.md]:
```suggestion
Component tests are run using [Cypress] tests on a single [Storybook] story. To load the component, call `cy.visit()` on the URL for that particular story. You can find the URL by opening that story in Storybook, clicking the "Open canvas in new tab" icon in the upper right (the external link icon), and copying the URL from that tab. Now you can write your component tests using Cypress' API. Note that you do *not* need to import Chai or Sinon—these are included in Cypress. Here is a small example test:
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -48,153 +66,481 @@ This project is driven by the needs of documentary and descriptive linguists, an
...
+Cypress tests can either be run programmatically (from the command line), or by using an interactive interface which allows you to watch the tests interact with the app, rerun, and debug those tests.
+
+You can run the various types of tests with the following commands:
+
+| Test Type   | Extension         | Interface    | Command (with `npm run`)          |
+| ----------- | ----------------- | ------------ | --------------------------------- |
+| unit        | `.unit.js`        | interactive  | —                                 |
+| unit        | `.unit.js`        | programmatic | `test:unit`                       |
+| component   | `.component.js`   | interactive  | `cypress-ct`                      |
+| component   | `.component.js`   | programmatic | `test:component`                  |
```
Comment (vadekh) [inline .github/CONTRIBUTING.md:396]: Looks like this command has been/will be removed as well. via #236

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -48,153 +66,481 @@ This project is driven by the needs of documentary and descriptive linguists, an
...
+
+Cypress tests can either be run programmatically (from the command line), or by using an interactive interface which allows you to watch the tests interact with the app, rerun, and debug those tests.
+
+You can run the various types of tests with the following commands:
+
+| Test Type   | Extension         | Interface    | Command (with `npm run`)          |
+| ----------- | ----------------- | ------------ | --------------------------------- |
+| unit        | `.unit.js`        | interactive  | —                                 |
+| unit        | `.unit.js`        | programmatic | `test:unit`                       |
+| component   | `.component.js`   | interactive  | `cypress-ct`                      |
```
Comment (vadekh) [inline .github/CONTRIBUTING.md:395]: Looks like this command has been/will be removed.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -48,153 +66,481 @@ This project is driven by the needs of documentary and descriptive linguists, an
...
+| component   | `.component.js`   | programmatic | `test:component`                  |
+| integration | `.integration.js` | interactive  | `cypress-it`                      |
+| integration | `.integration.js` | programmatic | `test:integration`                |
+| E2E         | `.e2e.js`         | interactive  | `cypress-e2e`                     |
+| E2E         | `.e2e.js`         | programmatic | `test:e2e`                        |
+| performance | —                 | interactive  | [Chromium dev tools][lh-devtools] |
+| performance | —                 | programmatic | `test:perf`                       |
+
+Alternatively, you can open both Storybook + Cypress manually, which is sometimes easier for development. First run `npm run storybook`, and then in a separate terminal run `npm run cypress-ct-open`, `npm run cypress-e2e-open`, or `npm run cypress-it-open` depending on whether you want to run component, E2E, or integration tests respectively.
```
Comment (vadekh) [inline .github/CONTRIBUTING.md:404]: Again, `cypress-ct` is being removed so should it still appear here?

**Review by vadekh — CHANGES_REQUESTED** [review]
Just a missing space and a few things for consistency with the #236 changes to `package.json`

**Comment by vadekh** [comment]
I think this closes #236

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (vadekh) [reply .github/CONTRIBUTING.md:395]: I see now that this reflects changes in #249, so maybe we should revisit the consistency of this table after that PR is merged.

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (dwhieb) [reply .github/CONTRIBUTING.md:396]: This will supersede #236 now.

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (dwhieb) [reply .github/CONTRIBUTING.md:395]: Yes good call!

**Review by vadekh — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -48,153 +66,482 @@ This project is driven by the needs of documentary and descriptive linguists, an
...
+| `.storybook/` | Configuration for [Storybook].                                                                                                                         |
+| `build/`      | Scripts to build the production version of the app from the `src/` files.                                                                              |
+| `dist/`       | Production code for the app. The contents of this folder are deployed to the production server on release, and a staging server on pull requests.      |
+| `docs/`       | Developer documentation. The contents of this folder are deployed to https://developer.digitallinguistics/app.                                         |
+| `src/`        | Source code for the app. Test files should live alongside their source components. See [App Structure](#app-structure) below for details.              |
+| `test/`       | Configuration code and fixtures for tests. Test specs should _not_ be placed here unless they are tests having to do with the development environment. |
```
Comment (TongJ05) [inline .github/CONTRIBUTING.md]: Later in the document(line 233), the document says unit tests for a single component is placed in the component folder, not in test/ folder. So maybe adding something like "tests about components are not placed here" is clearer.

**Review by TongJ05 — APPROVED** [review]
This is clear for me and easy to understand! The only thing to improve is the introduction to test organization addressed in the comment.

**Pull request merged**